snuba/admin/views.py

```diff
@@ -1138,6 +1141,47 @@ def dlq_replay() -> Response:
     return make_response(loaded_instruction.to_bytes().decode("utf-8"), 200)
 
 
+@application.route("/rpc_endpoints", methods=["GET"])
+@check_tool_perms(tools=[AdminTools.RPC_ENDPOINTS])
+def list_rpc_endpoints() -> Response:
+    return Response(
+        json.dumps(RPCEndpoint.list_all_endpoint_names()),
+        200,
+        {"Content-Type": "application/json"},
+    )
+
+
+@application.route("/rpc_execute/<endpoint_name>", methods=["POST"])
+@check_tool_perms(tools=[AdminTools.RPC_ENDPOINTS])
+def execute_rpc_endpoint(endpoint_name: str) -> Response:
+    endpoint_class = RPCEndpoint.get_from_name(endpoint_name, "v1alpha")
+    if not endpoint_class:
+        return Response(
+            json.dumps({"error": f"Unknown endpoint: {endpoint_name}"}),
+            404,
+            {"Content-Type": "application/json"},
+        )
+
+    body = request.json
+
+    try:
+        request_proto = Parse(json.dumps(body), endpoint_class.request_class()())
```

Comment:
it seems RPCEndpoint.request_class can return any type, but this suggests it must always be `google.protobuf.message.Message`
I would have expected this to cause a typing issue

Reply:
Oh thank you that makes more sense